src/main/java/com/adobe/cq/commerce/graphql/client/impl/GraphqlClientAdapterFactory.java

```diff
@@ -75,17 +80,11 @@ protected void unbindGraphqlClient(GraphqlClient graphqlClient, Map<?, ?> proper
 
         Resource res = (Resource) adaptable;
 
-        // Get cq:graphqlClient property from ancestor pages
-        InheritanceValueMap properties;
-        Page page = res.getResourceResolver().adaptTo(PageManager.class).getContainingPage(res);
-        if (page != null) {
-            properties = new HierarchyNodeInheritanceValueMap(page.getContentResource());
-        } else {
-            properties = new ComponentInheritanceValueMap(res);
-        }
+        ConfigurationBuilder configBuilder = res.adaptTo(ConfigurationBuilder.class);
+        ValueMap properties = configBuilder.name(CONFIG_NAME).asValueMap();
 
-        String identifier = properties.getInherited(GraphqlClientConfiguration.CQ_GRAPHQL_CLIENT, String.class);
-        if (identifier == null) {
+        String identifier = properties.get(GraphqlClientConfiguration.CQ_GRAPHQL_CLIENT, readFallbackConfiguration(res));
```

Comment:
I would suggest to call readFallbackConfiguration() only if needed. Currently it will be called every time for passing its return value to properties.get() which will be always ignored if the context aware configuration is available.

Reply:
I agree with Levente, and we should have a dedicated test for each configuration option (right now you only test the "fallback" option).